File: benchmarks/javascript/jobs/matrix_mul.js
function matrix_mul() {
  const n = 512;

  const A = new Array(n)
    .fill(0)
    .map((_, i) => new Array(n).fill(0).map(() => i * 0.01));

  const B = new Array(n)
    .fill(0)
    .map((_, i) => new Array(n).fill(0).map((_, j) => j * 0.02));

  const C = new Array(n).fill(0).map(() => new Array(n).fill(0));

  for (let i = 0; i < n; i++) {
    for (let k = 0; k < n; k++) {
      const aik = A[i][k];
      for (let j = 0; j < n; j++) {
        C[i][j] += aik * B[k][j];
      }
    }
  }

  return C[0][0];
}

module.exports = { matrix_mul };

File: benchmarks/python/jobs/matrix_mul.py
def matrix_mul():
    n = 512

    A = [[float(i * 0.01) for i in range(n)] for _ in range(n)]
    B = [[float(j * 0.02) for j in range(n)] for _ in range(n)]
    C = [[0.0] * n for _ in range(n)]

    for i in range(n):
        for k in range(n):
            aik = A[i][k]
            for j in range(n):
                C[i][j] += aik * B[k][j]

    return C[0][0]
